packages/cli/src/commands/init/init.ts

```diff
@@ -501,25 +477,44 @@ async function createProject(
-  // +------------------------------------------------------------------+----------+--------------+
-  // | --template @react-native-community/template@0.75.1 --version 0.75| 0.75.1   | 0.75.x       |
-  // +------------------------------------------------------------------+----------+--------------+
+  // +==================================================================+==========+===================+
+  // | Arguments                                                        | Template |   React Native    |
+  // +==================================================================+==========+===================+
+  // | <None>                                                           | 0.74.x   | 0.74.5 (latest)   |
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --version next                                                   | 0.75.x   | 0.75.0-rc.1 (next)|
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --version 0.75.0                                                 | 0.75.x   | 0.75.0            |
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --template @react-native-community/template@0.75.1               | 0.75.1   | latest            |
+  // +------------------------------------------------------------------+----------+-------------------+
+  // | --template @react-native-community/template@0.75.1 --version 0.75| 0.75.1   | 0.75.x            |
+  // +------------------------------------------------------------------+----------+-------------------+
   //
   // 1. If you specify `--version 0.75.0` and `@react-native-community/template@0.75.0` is *NOT*
   // published, then `init` will exit and suggest explicitly using the `--template` argument.
   //
   // 2. `--template` will always win over `--version` for the template.
   //
   // 3. For version < 0.75, the template ships with react-native.
-  const templateUri = await createTemplateUri(options, version);
+  const v = semver.parse(version);
+  if (v == null) {
+    throw new Error(`--version=${version} isn't valid SEMVER`);
+  }
+  // This will use ~ to grab the latest patch version, so pinning to zero to make this clear.
+  //
+  // Because we allow for many PATCHes to occur for a version of react-native, it's not possible to know
+  // (without inspecting the template's npm metadata) which template works with which react-native version.
+  // We are guaranteed that the latest version of react-native  will work with the latest version of the
+  // template on that version.
+  //
+  // Example:
+  //
+  // Suppose react-native 0.75.0 was released, and 0.74.8 was the previous version. If I ran
+  // `init --version 0.74.8` it would be guaranteed to work with the latest version of the template
+  // matching the MAJOR.MINOR, e.g. 0.74.21.
+  const templateVersion = `${v.major}.${v.minor}.0`;
```

Comment:
This should be moved inside `createTemplateUri` and moved after `if (options.platformName) {` check